app/connectors/CompanyHouseConnector.scala

```diff
@@ -36,7 +38,11 @@ class CompanyHouseConnector @Inject()(appConfig: FrontendAppConfig, http: HttpCl
   }
   def validateCRN(data: CompanyDetails)(implicit ec: ExecutionContext): Future[Option[Boolean]] = {
     implicit val hc = HeaderCarrier().withExtraHeaders(("Authorization", appConfig.companyHouseRequestAuth))
-    http.GET(appConfig.companyHouseRequestUrl + data.companyReferenceNumber)
+    val httpClient =
+      if (configuration.getOptional[String]("proxy.proxyRequiredForThisEnvironment").contains("true"))
```

Comment:
For consistency in the service I feel like this should be another lazy val in the `FrontendAppConfig` and as it's a boolean you can use `.get[Boolean]` instead of reading it as a string to check it.
That also means we wouldn't need to inject the configuration in this connector too